pkg/scheduler/queue/tenant_queues.go

```diff
@@ -36,6 +36,12 @@ func (s querierIDSlice) Search(x QuerierID) int {
 	return sort.Search(len(s), func(i int) bool { return s[i] >= x })
 }
 
+type tenantRequest struct {
+	tenantID    TenantID
+	req         Request
+	maxQueriers int
```

Comment:
From the perspective of the queue broker, is the latest max queriers value part of the request to be enqueued, or just some information that is updated at the same time as enqueuing a new request? It looks like the latter: it's only used when enqueuing a new request. If so, I'm wondering if `maxQueriers` shouldn't be a member of `tenantRequest` and should instead be a parameter to `enqueueRequestBack`.